telemetry/prometheus/metrics.go

```diff
@@ -152,16 +154,26 @@ func (p *metrics) IncMonotonic(name string, tags ...string) {
 
 	name = forceValidName(name)
 	labels, labelValues := parseTagsToLabelPairs(tags)
-	counterVec, exists := p.counterVecs[name]
-	if !exists {
-		counterVec = prometheus.NewCounterVec(prometheus.CounterOpts{
-			Name:      name,
-			Namespace: p.cfg.Namespace,
-			Subsystem: p.cfg.Subsystem,
-			Help:      name + " counter",
-		}, labels)
-		prometheus.MustRegister(counterVec)
-		p.counterVecs[name] = counterVec
+	if counterVec, exists := p.counterVecs.Get(name); exists {
+		counterVec.WithLabelValues(labelValues...).Inc()
+	}
+
+	counterVec := prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name:      name,
+		Namespace: p.cfg.Namespace,
+		Subsystem: p.cfg.Subsystem,
+		Help:      name + " counter",
+	}, labels)
+	if err := prometheus.Register(counterVec); err != nil {
```

Comment:
Better to use Register instead of MustRegister - so if we got AlreadyRegisteredError we can just retrieve the already registered metrics from the Error